### PR TITLE
Add logging to scala worker

### PR DESCRIPTION
**Pull request by yyuan-lucid**

### CODE REVIEW

File: src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala [hunk 1]
```diff
@@ -69,13 +69,17 @@ trait WorkerMain[S] {
...
             val request = WorkerProtocol.WorkRequest.parseDelimitedFrom(stdin)
+            if (request == null) {
+              return
+            }
             val args = request.getArgumentsList.toArray(Array.empty[String])
 
             val outStream = new ByteArrayOutputStream
             val out = new PrintStream(outStream)
             val requestId = request.getRequestId()
+            System.out.println(s"A WorkRequest $requestId is received. Args: $args")
```
Comment (JaredNeil) [inline src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala]:
```suggestion
            System.out.println(s"WorkRequest $requestId received with args: $args")
```

File: src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala [hunk 2]
```diff
@@ -90,11 +94,13 @@ trait WorkerMain[S] {
               case Success(code) => {
                 out.flush()
                 writeResponse(requestId, outStream, code)
+                System.out.println(s"Work future successfully completed. A WorkResponse $requestId was sent.")
```
Comment (JaredNeil) [inline src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala]:
```suggestion
                System.out.println(s"WorkResponse $requestId sent with code $code")
```

File: src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala [hunk 3]
```diff
@@ -90,11 +94,13 @@ trait WorkerMain[S] {
...
               }
               case Failure(e) => {
                 e.printStackTrace(out)
                 out.flush()
                 writeResponse(requestId, outStream, -1)
+                System.err.println(s"Work future failed. A WorkResponse $requestId was sent.")
```
Comment (JaredNeil) [inline src/main/scala/higherkindness/rules_scala/common/worker/WorkerMain.scala]:
```suggestion
                System.err.println(s"Uncaught exception in Future while proccessing WorkRequest $requestId:")
                e.printStackTrace(System.err)
```

**Review by JaredNeil — CHANGES_REQUESTED** [review]

**Review by JaredNeil — APPROVED** [review]

**Comment by JaredNeil** [comment]
Were you able to test this and see the logs?

**Comment by yyuan-lucid** [comment]
Yup. Also confirmed the NullPointerException went away with the fix.
image

**Comment by JaredNeil** [comment]
Oh, can you print the actual strings in `args`?

**Comment by yyuan-lucid** [comment]
image

**Pull request merged**